spark/src/main/scala/org/apache/spark/sql/delta/commands/VacuumCommand.scala

```diff
@@ -79,6 +80,26 @@ object VacuumCommand extends VacuumCommandImpl with Serializable {
        """.stripMargin)
   }
 
+  private def getValidPredicateFilterExpression(
+      spark: SparkSession,
+      partitionColumns: Seq[String],
+      userPartitionPredicates: Seq[String]
+      ): Seq[Expression] = {
+    require(
+      (partitionColumns.isEmpty && userPartitionPredicates.isEmpty)
+      ||partitionColumns.nonEmpty,
```

Comment:
Before my first coffee, so trying to work out the logic of this `require` from a truth table:
|         | partitionColumns.isEmpty | partitionColumns.nonEmpty |
| ----- | -------------- | --------------- |
| userPartitionPredicates.isEmpty | Pass | Pass |
| userPartitionPredicates.nonEmpty | Fail | Pass |

Perhaps this slight change makes the intent more clear?
```
  // A WHERE clause in a VACUUM statement is valid, if:
  //  - You have no WHERE clause
  //  - You have partition columns
  userPartitionPredicates.isEmpty || partitionColumns.nonEmpty
```

Reply:
yes, thanks for pointing redundant `partitionColumns.isEmpty` check. I will remove it.

Reply:
@watfordkcf I have updated the requirement check. Thank you.